### DISCUSSION TITLE
unclear how to ignore npm dependencies below 1.0.0

### DISCUSSION THREAD

**Discussion by iamstarkov** [renovatebot/renovate#23761] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

No PRs which bump dependencies to versions below `1.0.0`.
Separate PRs to bump a version once dependency reaches `1.0.0`.

all of it while bundling non-major devdeps, non-major (prod) deps in two separate PRs and one PR for every major/breaking bump.

The following config is separating bumps into three groups, I've talked about in the prev sentence. Previous renovate app version didnt give the heads about the error in this config, it just never respected `allowedVersions`. In order to ask a question I've created a [demo repo with and then a fresh new renovate installation gave me an error](https://github.com/iamstarkov/renovate-allowed-versions-confusion/issues/1): `packageRules[0]: packageRules cannot combine both matchUpdateTypes and allowedVersions.`

```
{
  "extends": [
    "config:base",
    "github>whitesource/merge-confidence:beta"
  ],
  "packageRules": [
    {
      "labels": ["devdeps"],
      "matchDatasources": ["npm"],
      "matchDepTypes": ["devDependencies"],
      "allowedVersions": ">=1",
      "matchUpdateTypes": ["patch", "minor"],
      "groupName": "developer dependencies (non-major)",
      "groupSlug": "devdeps"
    },
    {
      "labels": ["proddeps"],
      "matchDatasources": ["npm"],
      "allowedVersions": ">=1",
      "matchDepTypes": ["dependencies"],
      "matchUpdateTypes": ["patch", "minor"],
      "groupName": "production dependencies (non-major)",
      "groupSlug": "proddeps"
    },
    {
      "labels": ["breaking"],
      "registryUrls": [
        "https://nordnet.jfrog.io/artifactory/api/npm/npm-virtual/"
      ],
      "matchDatasources": ["npm"],
      "matchDepTypes": ["dependencies"],
      "matchUpdateTypes": ["major"]
    }
  ]
}
```

The question is im not quite sure how to achieve what I want. because I like the grouping, but i wanted to exclude dangerous sub 1.0.0 packages until they graduated to 1.0.0.

### What you tried so far.

I've tried `allowedVersions` and `matchUpdateTypes` options together and they apparently dont work

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
I think what you want is https://github.com/renovatebot/renovate/discussions/21637#discussioncomment-5692218 ?

i.e. use matchCurrentVersion to exclude any dependency starting with `0.`

**Reply by iamstarkov** [reply]
@rarkins thank you, that's what I was looking for but apparently overlooked
